app/helpers/newsletter_helper.rb

```diff
@@ -0,0 +1,9 @@
+module NewsletterHelper
+  def available_actions(newsletter)
+    if newsletter.draft? 
+      { actions: [:edit, :destroy] }
+    else
+      { actions: [:destroy] }
+    end
+  end
```

Comment:
With these changes we remove the link to the `edit` action, but the action is still accessible by manually typing the URL on the browser :thinking:.

How about defining abilities in `app/models/abilities/administrator.rb`, changing the `Admin::NewslettersController` so it uses the `load_and_authorize_resource!` method, and using `can?` here? Maybe something like:

```suggestion
  def available_actions(newsletter)
    [:edit, :destroy].select { |action| can?(action, newsletter) }
  end
```